Frameworks/UIKit/UITextField.mm

```diff
@@ -1183,348 +891,198 @@ - (void)setContentVerticalAlignment:(UIControlContentVerticalAlignment)alignment
-        __strong UITextField* strongSelf = weakSelf;
-        if (strongSelf) {
-            dispatch_async(dispatch_get_main_queue(), ^{
-                [strongSelf sendActionsForControlEvents:UIControlEventEditingChanged];
-            });
-        }
-    });
+    if (dismissKeyboard) {
+        [self resignFirstResponder];
+    }
 }
 
 // Main entrance to initialize TextField
-- (void)_initUITextField:(RTObject*)xamlElement {
-    self->_secureModeLock = [NSRecursiveLock new];
+- (void)_initUITextField:(BOOL)secureTextEntry {
+    _textField = [self _winrtXamlElement].as<FrameworkElement>();
 
-    // creating dummy button and hidden view so that it can be used to steal/kill the focus for this UITextField
-    self->_dummyButton = Controls::Button();
-    self->_dummyButton.Visibility(Visibility::Visible);
-    self->_dummyButton.IsEnabled(true);
-    self->_dummyButton.IsTabStop(true);
+    if (!_textField) {
+        // Definitely didn't receive any supported backing XAML elements
+        FAIL_FAST();
+    }
 
-    _hiddenView = [[_UIHiddenButtonView alloc] initWithFrame:CGRectZero xamlElement:objcwinrt::to_rtobj(_dummyButton)];
-    [self addSubview:_hiddenView];
+    // Initially both controls can be lazy loaded, we instantiate the appropriate control
+    // at runtime based on the secureTextEntry value
+    XamlControls::SetTextFieldSecureTextEntryValue(_textField, secureTextEntry);
+    __weak UITextField* weakSelf = self;
 
-    if (self->_secureTextMode) {
-        [self _initPasswordBox:nil];
-    } else {
-        [self _initTextBox:xamlElement];
-    }
+    XamlControls::TextFieldRegisterEventHandlers(_textField,
+                                                 // Text Changed event
+                                                 objcwinrt::callback([weakSelf](const WF::IInspectable& sender, const RoutedEventArgs&) {
+                                                     __strong UITextField* strongSelf = weakSelf;
+                                                     if (strongSelf) {
+                                                         dispatch_async(dispatch_get_main_queue(), ^{
+                                                             [strongSelf sendActionsForControlEvents:UIControlEventEditingChanged];
+                                                             [[NSNotificationCenter defaultCenter]
+                                                                 postNotificationName:UITextFieldTextDidChangeNotification
+                                                                               object:strongSelf];
+                                                         });
+                                                     }
+                                                 }),
+
+                                                 // Got focus event
+                                                 objcwinrt::callback([weakSelf](const WF::IInspectable& sender, const RoutedEventArgs&) {
```

Comment:
>objcwinrt::callback( [](start = 49, length = 20)

I don't think you need these, because the event callback already pushes/pops an autoreleasepool